### PR TITLE
[ADD] gs1: init

### PR DESCRIPTION
**Pull request by oerp-odoo**
Initial GS1 implementation

### CODE REVIEW

File: gs1/gs1.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+# TODO: replace this with dataclass, once we can drop 3.6 support.
+BARCODE_LENGTHS = {
+    "gtin-8": 7,
+    "gtin-12": 11,
+    "gtin-13": 12,
+    "gtin-14": 13,
+    "gsin": 16,
+    "sscc": 17,
+
+}
```
Comment (dangirasv) [inline gs1/gs1.py]: Unnecessary empty line?
```suggestion
    "sscc": 17,
}
```

**Review by dangirasv — APPROVED** [review]
Is `tests/__init__.py` file needed considering it's an empty file? No need to import `test_gs1.py` there?

**Comment by oerp-odoo** [comment]
> Is `tests/__init__.py` file needed considering it's an empty file? No need to import `test_gs1.py` there?

Yes its needed, otherwise it won't recognize it as package. And no need to import as it is discovered automatically. Speaking of which, `__init__.py` should also be present in `gs1`. As I forgot to do that:)

File: gs1/gs1.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+    Additionally, Application Identifier and Extension are included at
+    the start of barcode. Can pass empty string, to omit that.
+    """
+    return _get_barcode("sscc", company_prefix, serial, app_id=app_id, ext=ext)
+
+
+def _get_barcode(
+    type_: str, company_prefix: str, serial: str, app_id=APP_ID, ext=EXT
+) -> str:
```
Comment (DarkoNikolovski) [inline gs1/gs1.py:88]: Is there any reason why the type of variables is defined and the output type of the method?

**Review by DarkoNikolovski — APPROVED** [review]

File: gs1/gs1.py (hunk 2 above)
Comment (oerp-odoo) [reply gs1/gs1.py:88]: What do you mean? You are asking why there are type hints?

**Pull request merged**